src/lib/components/wizard.component.ts

```diff
@@ -62,33 +67,95 @@ export class WizardComponent implements OnChanges, AfterContentInit {
   @ContentChildren(WizardStep)
   public wizardSteps: QueryList<WizardStep>;
 
+  /**
+   * Step Changed event that fired with the new current step
+   */
+  @Output()
+  stepChanged: EventEmitter<WizardStep> = new EventEmitter<WizardStep>();
```

Comment:
I believe it's a good idea to add a `stepChanged` output emitter.
I'm just not sure if we should emit the `WizardStep` object, to which the wizard changed.
The reason here is, that the `WizardStep` is meant as an wizard internal class, which provides a lot of fields which shouldn't be changed outside of the wizard (for example the field `selected` shouldn't be changed from outside the wizard).

In addition the output doesn't forward information such as the navigation direction, the source step etc, which is often also of interest.
I think it's better to brainstorm some more about which information the output should forward and how it should look like.

Reply:
Sounds good.
Are there maybe some other fields we want to pass to the receiver of the output emitter (just asking would be bad if we missed something)?

Reply:
I don’t think it’s necessary but you can provide source and isTrusted:


#### Source
source of the changed step: from where, is it from clicking next/prev/go-to step or by clicking the step title

#### isTrusted
Is the change came from our library:
- Next/Prev/Go To step directives
- step title
- or programmatically